app/assets/javascripts/services/miq_service.js

```diff
@@ -124,4 +124,23 @@ ManageIQ.angular.app.service('miqService', ['$timeout', '$document', '$q', funct
 
     return $q.reject(e);
   };
+
+  this.getProviderTenants = function(callback) {
+    return function(id) {
```

Comment:
Well, this is specific to 4 different controllers, but yeah, maybe a separate service would make sense.

@tzumainn Any idea what's common in those 4 controllers? (Thinking about naming, should that be a `CloudService` or a `NetworkingService` or... :))